### PR TITLE
add support for aliasing with `.as()` in ValueBuilder

### PR DESCRIPTION
**Pull request by nigelzor**
the Typescript typings claim that this method already exists.

### CODE REVIEW

**Comment by lehni** [comment]
@nigelzor this looks great, but do you have an idea why no checks were run on your branch? https://github.com/Vincit/objection.js/pull/2346/checks

Is there a way to retrigger it from your end?

**Comment by nigelzor** [comment]
I can't trigger the checks manually because I don't have write access to this repo.

If I'm understanding [the docs](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) correctly, a maintainer needed to approve the triggered checks when I made the PR.

**Review by lehni — APPROVED** [review]

File: lib/queryBuilder/ValueBuilder.js [hunk 1]
```diff
@@ -62,26 +63,41 @@ class ValueBuilder {
...
     if (this._toJson) {
-      bindings = JSON.stringify(this._value);
+      bindings.push(JSON.stringify(this._value));
       sql = '?';
     } else if (this._toArray) {
-      bindings = asArray(this._value).map((it) => buildArg(it, builder));
-      sql = `ARRAY[${bindings.map(() => '?').join(', ')}]`;
+      const valueAsArray = asArray(this._value);
+      bindings.push(...valueAsArray.map((it) => buildArg(it, builder)));
+      sql = `ARRAY[${valueAsArray.map(() => '?').join(', ')}]`;
```
Comment (lehni) [inline lib/queryBuilder/ValueBuilder.js:85]: Let's name it simply `values` instead of `valueAsArray`, that's clear enough.

**Comment by nigelzor** [comment]
I've rebased. The workflow is now waiting for your approval.

**Comment by lehni** [comment]
The tests passed but the check failed at the very end due to 7769d51e144252756eda78a2776014f69e14b19d. I've fixed that an merged it now. Thanks!

**Comment by lehni** [comment]
All green now: https://github.com/Vincit/objection.js/actions/runs/4703659390/jobs/8342420749

**Pull request merged**